integration/helpers/usercreds.go

```diff
@@ -109,17 +108,18 @@ func GenerateUserCreds(req UserCredsRequest) (*UserCreds, error) {
 		ttl = time.Hour
 	}
 
-	priv, err := testauthority.New().GeneratePrivateKey()
+	key, err := client.GenerateRSAKey()
```

Comment:
Any reason you can't use the `testauthority` here? Didn't we just improve test speed and stability by reusing keypairs instead of generating hundreds of them during test runs?

Reply:
I believe you're referring to [this change](https://github.com/gravitational/teleport/pull/19507), which makes `testauthority` just call `native.GenerateKeyPair` under the hood, so it's functionally equivalent. 

See also this comment in the PR description:

> We could probably refactor the code in many tests now to make it simpler, but I want to keep this change as simple as possible. Otherwise, backporting bigger changes to older branches takes a long time.

Reply:
`testauthority` uses this new pool that you've mentioned already. I want to clean our testing APIs after v12 release.